src/Container/OsRelease.hs

```diff
@@ -0,0 +1,238 @@
+-- | Parses system information identifier for linux based operating systems.
+--
+-- Typically such information is stored in following paths:
+--
+--  * /etc/os-release (common)
+--  * /usr/lib/os-release (common)
+--  * /etc/initrd-release
+--  * /usr/lib/extension-release.d/extension-release.IMAGE
+--
+-- The format of os-release is a newline-separated list of environment-like
+-- shell-compatible variable assignments.
+--
+-- ## Example
+-- ----------
+--
+-- NAME="Alpine Linux"
+-- ID=alpine
+-- VERSION_ID=3.15.4
+-- PRETTY_NAME="Alpine Linux v3.15"
+-- HOME_URL="https://alpinelinux.org/"
+-- BUG_REPORT_URL="https://bugs.alpinelinux.org/"
+--
+-- ## Fields of Interest
+-- ---------------------
+--
+-- ID
+--
+--  A lower-case string (no spaces or other characters outside of 0–9, a–z, ".", "_" and "-") identifying the operating system,
+--  excluding any version information and suitable for processing by scripts or usage in generated filenames.
+--  If not set, a default of "ID=linux" may be used. Note that even though this string may not include characters
+--  that require shell quoting, quoting may nevertheless be used.
+--
+--
+-- VERSION_ID
+--
+--  A lower-case string (mostly numeric, no spaces or other characters outside of 0–9, a–z, ".", "_" and "-")
+--  identifying the operating system version, excluding any OS name information or release code name, and
+--  suitable for processing by scripts or usage in generated filenames. This field is optional.
+--
+-- ## Spec
+-- --------
+--
+-- * Variable assignment values are enclosed in double or single quotes if they include spaces, semicolons or
+--   other special characters outside of A–Z, a–z, 0–9.
+--
+-- * Shell special characters ("$", quotes, backslash, backtick) must be escaped with backslashes,
+--   following shell style. All strings should be in UTF-8 encoding, and non-printable characters
+--   should not be used. Concatenation of multiple individually quoted strings is not supported.
+--
+-- * Lines beginning with "#" are treated as comments.
+--
+-- * Blank lines are permitted and ignored.
+module Container.OsRelease (
```

Comment:
It would be good to simplify, but I can't imagine it will have a high priority given workload; regardless will add as a tech debt ticket, so it's in backlog